sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy

```diff
@@ -58,38 +32,34 @@ import static java.util.Objects.requireNonNull
  * @since 1.0.0
  *
  */
+@CompileStatic
 class DependencyManager {
-    private VerticalLayout portletView
     private ConfigurationManager configurationManager
     private final PortalUser portalUser
-    private final NotificationHandler notificationHandler
-
-    private LoadProjectsDataSource loadProjectsDataSource
-    private LastChangedDateDataSource lastChangedDateDataSource
-    private CountSamplesDataSource countSamplesDataSource
-    private GetSamplesInfoDataSource getSamplesInfoDataSource
-    private DownloadSamplesDataSource downloadSamplesDataSource
-    private SubscriptionDataSource subscriptionDataSource
-    private SubscribedProjectsDataSource subscribedProjectsDataSource
-    private SampleStatusDataSource sampleStatusDataSource
-
-    private ResourceService<Project> projectResourceService
-    private ResourceService<StatusCount> statusCountService
-    private NotificationService notificationService
+
     private Subscriber subscriptionUser
 
-    DependencyManager(PortalUser user) {
+    private static final int PROJECT_LOADING_THREAD_COUNT = 2
+    private static final int SAMPLE_LOADING_THREAD_COUNT = 2
+    private final ExecutorService projectLoadingExecutor = Executors.newFixedThreadPool(PROJECT_LOADING_THREAD_COUNT)
+    private final ExecutorService sampleLoadingExecutor = Executors.newFixedThreadPool(SAMPLE_LOADING_THREAD_COUNT)
```

Comment:
NIT: Is this something that could be moved to a settings file?

Reply:
good idea

Reply:
The only problem is, that we do not have a settings file in this project except the portlet properties. Thus, I will not change it.